daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesTextNumber.scala

```diff
@@ -46,8 +45,249 @@ case class ConvertTextCombinator(e: ElementBase, value: Gram, converter: Gram)
   override lazy val unparser = new ConvertTextCombinatorUnparser(e.termRuntimeData, value.unparser, converter.unparser)
 }
 
-case class ConvertTextNumberPrim(e: ElementBase)
-  extends Terminal(e, true) {
+// This is a separate object for unit testing purposes.
```

Comment:
I should have made this scaladoc.
I should have said "This object exists to expose some code for convenient unit testing that would otherwise be inaccessible."  It's not that the whole object can move into src/test. It's just there so we can do a bunch of unit tests on the regexs it contains.

Reply:
Addressed in https://github.com/apache/daffodil/pull/910